### ISSUE TITLE
DLT-based PnP is not helpful for training

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#2183]
### Describe the bug

According to [recent paper from Snap Inc](https://arxiv.org/pdf/2301.11326.pdf), kornia differentiable PnP does not work well

"Different PnP. We tested several different PnP imple- mentations. We found that implementations based on declarative layers [8,16] are extremely slow, and using them in our setting would have been unfeasible. We also tested an implementation from the Kornia Library [48] that is based on DLT. However, it did not produce any meaningful results and produced divergence of the model. Our final choice was the EPnP [28] implementation from Pytorch3D [44]. However, we would like to note that it was only working in PyTorch 10.1 and not PyTorch 11, where it was not converging. We discovered the problem was the initially unstable gradients of the pinverse function in the newer version. We think that this instability can be solved with better initialization for the 2d points, however we left this investigation for future work."

<img width="438" alt="image" src="https://user-images.githubusercontent.com/4803565/215055250-6f261388-5770-4329-94fb-ad0d5f33b986.png">

### Reproduction steps

```bash
1.
2.
3.
...
```

### Expected behavior

n/a

### Environment

```shell
n/a
```

**Comment by edgarriba** [comment]
If anyone is interested in this, feel free to re-open the ticket. Thanks!
